src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -174,6 +174,13 @@ public function getFlashes($types = null)
             return $session->getFlashBag()->get($types);
         }
 
-        return array_intersect_key($session->getFlashBag()->all(), array_flip($types));
+        $result = array();
+        foreach ($types as $type) {
+            if ($value = $session->getFlashBag()->get($type)) {
```

Comment:
@xabbuh , in fact, flashBag get method returns an empty array by default. Should I simply delete the if condition ? ^^